app/web/src/store/component_attributes.store.ts

```diff
@@ -62,59 +77,90 @@ export const useComponentAttributesStore = (componentId: ComponentId) => {
           values: null as PropertyEditorValues | null,
         }),
         getters: {
-          currentValueForValueId:
-            (state) =>
-            (valueId: string): PropertyEditorValue | undefined =>
-              state.values?.values[valueId],
-          // puts the schema, validations, values all together in a format used by the property editor
-          editorContext: (state) => {
-            const { schema, validations, values } = state;
-            if (!schema || !validations || !values) return undefined;
-
-            // previously called hackAwayTheZeroElementOfContainers - not entirely clear what it's doing
-            // can likely refactor how we store/retrieve the data so we wont need this...
-            const filteredChildValues: { [key: string]: Array<string> } = {};
-
-            for (const [parentValueId, childValuesIds] of Object.entries(
-              values.childValues,
-            )) {
-              const parentValue = values.values[parentValueId];
-              if (!parentValue) {
-                // If we don't find a value, then don't filter and continue
-                filteredChildValues[parentValueId] = childValuesIds;
-                continue;
-              }
-              const parentProp = schema.props[parentValue.propId];
-              if (!parentProp) {
-                // If we don't find a prop, then don't filter and continue
-                filteredChildValues[parentValue.id] = childValuesIds;
-                continue;
-              }
-
-              if (parentProp.kind === "array" || parentProp.kind === "map") {
-                filteredChildValues[parentValue.id] = childValuesIds.filter(
-                  (childValueId) => {
-                    const childValue = values.values[childValueId];
-                    if (childValue && _.isNull(childValue.key)) {
-                      // If we don't find a value, then don't filter it out
-                      return false;
-                    } else {
-                      return true;
-                    }
-                  },
-                );
-              } else {
-                filteredChildValues[parentValue.id] = childValuesIds;
-              }
+          // recombine the schema + values + validations into a single nested tree that can be used by the attributes panel
+          attributesTree: (state): AttributeTreeItem | undefined => {
```

Comment:
I hope this should be much easier to follow whats happening... As this stabilizes, we may want to fetch the data from the backend in a shape more like this, but we'll see!